### PR TITLE
Add a search engine example

### PR DESCRIPTION
**Pull request by wbamberg**

### CODE REVIEW

File: discogs-search/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## What it shows
+
+How to use the [`chrome_settings_overrides`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/chrome_settings_overrides) manifest key to define a new search engine.
+
+Note that Chrome mandates the `is_default` property, and `encoding` must also be included unless `prepopulated_id` is set. Since none of these properties are yet supported in Firefox, it's not possible to write a single manifest.json that works in both browsers. So this example includes three manifest*.json files:
+
+* manifest-chrome.json: works in Chrome. Rename this to manifest.json to use Chrome.
+* manifest-firefox.json: works in Firefox.
+* manifest.json: the default. The same as manifest-firefox.json
```
Comment (andymckay) [inline discogs-search/README.md]: At the same time the most simple and most complicated example in our repo :)

File: discogs-search/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+This add-on adds a search engine to the browser, that sends the search term to the [discogs.com](https://discogs.com) website.
+
+It also adds a keyword "disc", so you can type "disc Model 500" and get the discogs search engine without having to select it.
+
+## What it shows
+
+How to use the [`chrome_settings_overrides`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/chrome_settings_overrides) manifest key to define a new search engine.
+
+Note that Chrome mandates the `is_default` property, and `encoding` must also be included unless `prepopulated_id` is set. Since none of these properties are yet supported in Firefox, it's not possible to write a single manifest.json that works in both browsers. So this example includes three manifest*.json files:
```
Comment (andymckay) [inline discogs-search/README.md]: I'll file a bug to get those to not raise errors and throw a warning.

**Comment by andymckay** [comment]
I probably should have tried this whilst I did the review, but I'm not sure why we don't just use the Chrome manifest. Having those extra keys in the manifest doesn't stop it working at all.

screenshot 2017-06-23 14 34 42

**Comment by wbamberg** [comment]
Oh, that's good. I saw things in the console like "Error processing chrome_settings_overrides.search_provider.encoding: An unexpected property was found in the WebExtension manifest." and just assumed it had not worked. I'll fix it now.

**Pull request merged**